src/components/Beneficiaries/index.tsx

```diff
@@ -75,242 +43,207 @@ const defaultBeneficiary: Beneficiary = {
-
-// tslint:disable jsx-no-multiline-js
-class BeneficiariesComponent extends React.Component<Props, State> {
-    constructor(props: Props) {
-        super(props);
-        this.state = {
-            currentWithdrawalBeneficiary: defaultBeneficiary,
-            isOpenAddressModal: false,
-            isOpenConfirmationModal: false,
-            isOpenDropdown: false,
-            isOpenTip: false,
-            isOpenFailModal: false,
-        };
-    }
-
-    public componentDidMount() {
-        const { currency, beneficiaries, memberLevels } = this.props;
-        if (currency && beneficiaries.length) {
-            this.handleSetCurrentAddressOnUpdate(beneficiaries);
+type Props = OwnProps;
+
+const BeneficiariesComponent: React.FC<Props> = (props: Props) => {
+    const [currentWithdrawalBeneficiary, setWithdrawalBeneficiary] = React.useState(defaultBeneficiary);
+    const [isOpenAddressModal, setAddressModalState] = React.useState(false);
+    const [isOpenConfirmationModal, setConfirmationModalState] = React.useState(false);
+    const [isOpenFailModal, setFailModalState] = React.useState(false);
+    const [isOpenTip, setTipState] = React.useState(false);
+    const [isOpenDropdown, toggleDropdown] = React.useState(false);
+
+    const { currency, type, onChangeValue } = props;
+
+    const { formatMessage } = useIntl();
+    const dispatch = useDispatch();
+
+    /*    selectors    */
+    const beneficiaries = useSelector(selectBeneficiaries);
+    const beneficiariesAddData = useSelector(selectBeneficiariesCreate);
+    const beneficiariesAddSuccess = useSelector(selectBeneficiariesCreateSuccess);
+    const beneficiariesActivateSuccess = useSelector(selectBeneficiariesActivateSuccess);
+    const memberLevels = useSelector(selectMemberLevels);
+    const userData = useSelector(selectUserInfo);
+    const isMobileDevice = useSelector(selectMobileDeviceState);
+    /*    ---------    */
+
+    React.useEffect(() => {
+        if (currency && beneficiaries) {
+            handleSetCurrentAddressOnUpdate(beneficiaries);
         }
 
         if (!memberLevels) {
-            this.props.memberLevelsFetch();
-        }
-    }
-
-    public componentWillReceiveProps(nextProps: Props) {
-        const {
-            currency,
-            beneficiaries,
-            beneficiariesAddSuccess,
-            beneficiariesActivateSuccess,
-        } = this.props;
-
-        if ((nextProps.currency && nextProps.currency !== currency) ||
-            (nextProps.beneficiaries.length && nextProps.beneficiaries !== beneficiaries)) {
-            this.handleSetCurrentAddressOnUpdate(nextProps.beneficiaries);
+            dispatch(memberLevelsFetch());
         }
+    }, []);
+
+    const prevCurrency = usePrevious(currency);
+    const prevBeneficiaries = usePrevious(beneficiaries);
+    const prevBeneficiariesAddSuccess = usePrevious(beneficiariesAddSuccess);
+    const prevBenefiariesActivateSuccess = usePrevious(beneficiariesActivateSuccess);
```

Comment:
There is no need to have `usePrevious` hook. `useEffect` compares old and new properties.